connect/api/src/test/java/org/apache/kafka/connect/data/ValuesTest.java

```diff
@@ -327,6 +347,88 @@ public void shouldParseStringsWithMultipleDelimiters() {
         assertParsed("a { b c } d", "a ", "{", " b c ", "}", " d");
     }
 
+    @Test
+    public void shouldConvertTimeValues() {
+        java.util.Date current = new java.util.Date();
+        long currentMillis = current.getTime() % MILLIS_PER_DAY;
+        Long currentMillisL = new Long(currentMillis);
+
+        // java.util.Date - just copy
+        java.util.Date t1 = Values.convertToTime(Time.SCHEMA, current);
+        assertEquals(current, t1);
+
+        // java.util.Date as a Timestamp - discard the date and keep just day's milliseconds
+        t1 = Values.convertToTime(Timestamp.SCHEMA, current);
+        assertEquals(new java.util.Date(currentMillis), t1);
+
+        // ISO8601 strings - currently broken because tokenization breaks at colon
+
+        // Millis as string
+        java.util.Date t3 = Values.convertToTime(Time.SCHEMA, currentMillisL.toString());
+        assertEquals(currentMillisL, new Long(t3.getTime()));
```

Comment:
you can just compare the primitive longs, no need to do all this boxing to `Long` type

Reply:
Ah, yes. I'd not realised there was an assertEquals for long values.